From a93e7cc23fbb9a86b19b3f3aa545ee52be87cb95 Mon Sep 17 00:00:00 2001
From: Ilya Tagunov <tagunil@gmail.com>
Date: Mon, 20 Apr 2020 23:46:06 +0300
Subject: [PATCH 07/32] filter: grc: variable_band_pass_filter_taps: fix
 complex taps option

In the case of complex taps, firdes.complex_band_pass should be used
instead of firdes.band_pass.

Signed-off-by: Ilya Tagunov <tagunil@gmail.com>
---
 gr-filter/grc/variable_band_pass_filter_taps.block.yml | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/gr-filter/grc/variable_band_pass_filter_taps.block.yml b/gr-filter/grc/variable_band_pass_filter_taps.block.yml
index 1903c9af31..a2bd965e7b 100644
--- a/gr-filter/grc/variable_band_pass_filter_taps.block.yml
+++ b/gr-filter/grc/variable_band_pass_filter_taps.block.yml
@@ -36,7 +36,8 @@ parameters:
     label: Beta
     dtype: float
     default: '6.76'
-value: ${ firdes.band_pass(gain, samp_rate, low_cutoff_freq, high_cutoff_freq, width,win, beta) }
+value: ${ getattr(firdes, str(type))(gain, samp_rate, low_cutoff_freq, high_cutoff_freq,
+    width, win, beta) }
 
 templates:
     imports: from gnuradio.filter import firdes
-- 
2.20.1

